analyzers/src/SonarAnalyzer.CSharp/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -255,6 +255,9 @@ static SyntaxNode PushPathPositionForTuple(Stack<PathPosition> pathPositions, Tu
                     : null;
         }
 
+        public static bool IsInTopLevelMain(this SyntaxNode node) =>
+            node.AncestorsAndSelf().OfType<CompilationUnitSyntax>().FirstOrDefault()?.ChildNodes().OfType<GlobalStatementSyntax>().Any() ?? false;
```

Comment:
The node that is the `CompilationUnit` in the case of top-level statements should always be the root.
Is there a way to get directly the root and if yes - it is faster than getting the ancestors in a list? 
I mean something like this: `node.GetRoot().ChildNodes().OfType<GlobalStatementSyntax>().Any()`.

Reply:
Good point. The root should always be `CompilationUnit`. It can return `null` though so I've adjusted the code a bit.